website/docs/cdktf/debugging.mdx

```diff
@@ -0,0 +1,58 @@
+---
+page_title: Debugging
+description: >-
+  Turn on logging and troubleshoot problems in your CDKTF application.
+---
+
+# Debugging
+
+To debug your CDKTF application, we recommend first turning on logging. Then, determine whether the issue occurs when your configuration is synthesized or at runtime.
+
+## Logging
+
+Set the `CDKTF_LOG_LEVEL` environment variable to `debug`. This produces more detailed information about each execution step. If you want to create a bug report or share the result, write the logs to a file by setting `CDKTF_LOG_FILE_DIRECTORY` to the file path where you want CDKTF to write the `cdktf.log` file.
+
+## Synth-time issues
+
+When you run `cdktf synth`, CDKTF translates your CDKTF application code into a JSON Terraform configuration file. Your application has a synth-time issue if `cdktf synth` fails. Otherwise, assume the issue occurs at execution time.
+
+When you encounter a synth issue, you should first identify the construct that is causing the error. The error message thrown might hint you towards it, if not you will need to search for it. If you have a last version that successfully synthesized you can search through the new changes, otherwise you need to search through all changes. To search it you can repeatedly run `cdktf synth` and comment parts out until you find the error. The synthesized code does not need to make any logical sense since we don't plan to apply it.
+
+Once you find the construct causing the problem you need to try to isolate the problem. If you use variables in your constructs configuration, try to temporarily replace them with static values to ensure the problem is on the construct itself. If the problem persists try looking at the type to identfy missing or mistyped fields. Looking at the resource in the Terraform provider might help to understand the structure better as well.
+
+## Execution-time issues
+
+Execution-time issues can arise from misconfigured synthesized code or from the provider itself.
```

Comment:
```suggestion
Execution-time issues happen when Terraform cannot successfully provision infrastructure with the synthesized configuration file. They can occur when you misconfigure Terraform resources in your CDKTF application, resulting in a misconfigured or incomplete Terraform configuration. They can also occur because of issues with the infrastructure provider itself. 
```